Replicator/Replicator.cc

```diff
@@ -664,9 +664,19 @@ namespace litecore { namespace repl {
         if(!db) {
             return false;
         }
-        
-        _checkpointer.pendingDocumentIDs(db->useLocked(), callback);
-        return true;
+
+        try {
+            db->useLocked([this, callback](const Retained<C4Database>& db) {
```

Comment:
It should be OK to keep the shorter `db->useLocked()` mode ... it unlocks the db if there's an exception.
(If you found that it didn't, that's an important bug; let me know!)

Reply:
Use db->useLocked() will throw. However, I want to protect the situation when at the time calling db->useLocked(), it's in good state but when we use the db from the return, it is subject to other thread nulling it. By using this version, I will protect it because I am holding the mutex during entire time the db is used.